src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,177 @@
+
+        array_walk_recursive($structure, function (&$item) use (&$generators) {
+            // generators should be used but for better DX all kind of Traversable are supported
+            if ($item instanceof \Traversable && !$item instanceof \JsonSerializable) {
+                $generators[] = $item;
+                $item = self::PLACEHOLDER;
+            } elseif (self::PLACEHOLDER === $item) {
+                // if the placeholder is already in the structure it should be replaced with a new one that explode
+                // works like expected for the structure
+                $generators[] = $item;
+            }
+        });
+
+        $jsonEncodingOptions = \JSON_THROW_ON_ERROR | $this->getEncodingOptions();
+        $keyEncodingOptions = $jsonEncodingOptions & ~\JSON_NUMERIC_CHECK;
+
+        $jsonParts = explode('"'.self::PLACEHOLDER.'"', json_encode($structure, $jsonEncodingOptions));
+
+        foreach ($generators as $index => $generator) {
+            // send first and between parts of the structure
+            echo $jsonParts[$index];
+
+            if (self::PLACEHOLDER === $generator) {
+                // the placeholders already in the structure are rendered here
+                echo json_encode(self::PLACEHOLDER, $jsonEncodingOptions);
+
+                continue;
+            }
+
+            $count = 0;
+            $startTag = '[';
+            foreach ($generator as $key => $item) {
+                if (0 === $count) {
+                    // depending on the first elements key the generator is detected as a list or map
+                    // we can not check for a whole list or map because that would hurt the performance
+                    // of the streamed response which is the main goal of this response class
+                    if (0 !== $key) {
+                        $startTag = '{';
+                    }
+
+                    echo $startTag;
+                } else {
+                    // if not first element of the generic, a separator is required between the elements
+                    echo ',';
+                }
+
+                if ('{' === $startTag) {
+                    echo json_encode((string) $key, $keyEncodingOptions).':';
+                }
+
+                echo json_encode($item, $jsonEncodingOptions);
+                ++$count;
+
+                if (0 === $count % $this->flushSize) {
+                    flush();
+                }
+            }
+
+            echo '[' === $startTag ? ']' : '}';
+        }
+
+        // send last part of the structure
+        echo $jsonParts[array_key_last($jsonParts)];
+    }
+
+    /**
+     * Returns options used while encoding data to JSON.
+     */
+    public function getEncodingOptions(): int
+    {
+        return $this->encodingOptions;
+    }
+
+    /**
+     * Sets options used while encoding data to JSON.
+     *
+     * @return $this
+     */
+    public function setEncodingOptions(int $encodingOptions): static
```

Comment:
I did add this because it exist also for `JsonResponse` and wanted to be consistent: https://github.com/symfony/symfony/blob/58f6a33d13d0181f2ab335203c5f72f1c080e939/src/Symfony/Component/HttpFoundation/JsonResponse.php#L163